src/compiler/Parser/ParsedHierarchy/Expression/NumberExpression.ts

```diff
@@ -39,4 +45,14 @@ export class NumberExpression extends Expression {
   };
 
   public readonly toString = (): string => String(this.value);
+
+  public Equals(obj: ParsedObject): boolean {
```

Comment:
implementation was missing, it slightly differs from the original one because here we have subtypes, C# does it automatically.